sql/insert.go

```diff
@@ -60,19 +60,38 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		return nil, err
 	}
 
+	primaryIndex := tableDesc.Indexes[0]
+	primaryIndexKeyPrefix := encodeIndexKeyPrefix(tableDesc.ID, primaryIndex.ID)
+
 	b := client.Batch{}
+
 	for rows.Next() {
 		values := rows.Values()
 		if len(values) != len(cols) {
 			return nil, fmt.Errorf("invalid values for columns: %d != %d", len(values), len(cols))
 		}
-		indexKey := encodeIndexKeyPrefix(desc.ID, desc.Indexes[0].ID)
-		primaryKey, err := encodeIndexKey(desc.Indexes[0], colMap, values, indexKey)
+
+		primaryIndexKeySuffix, err := encodeIndexKey(primaryIndex, colMap, values, nil)
 		if err != nil {
 			return nil, err
 		}
+		primaryIndexKey := append(primaryIndexKeyPrefix, primaryIndexKeySuffix...)
+
+		// Write the secondary indexes.
+		for _, secondaryIndex := range tableDesc.Indexes[1:] {
+			secondaryIndexKeyPrefix := encodeIndexKeyPrefix(tableDesc.ID, secondaryIndex.ID)
+			secondaryIndexKey, err := encodeIndexKey(secondaryIndex, colMap, values, secondaryIndexKeyPrefix)
+			if err != nil {
+				return nil, err
+			}
+
+			// TODO:(tamird): check uniqueness constraints (primary and
+			// secondary). Probably will have to do some scans?
```

Comment:
I think you can factor out the secondary index key construction into a method here in insert or in keys/... and add some unittests for the key construction. That way the code in here is just inserting/deleting keys.